### PR TITLE
Fix multiple secrets with same certificate

### PR DESCRIPTION
**Pull request by nicksardo**
Sometimes we hit timeouts in the e2e test: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-ingress-gce-e2e/1719

The controller is leaking an SSL Certificate and the test cleanup gives the controller a lot of time to clean it... Though, it never does.
```log
...
I0412 10:38:57.917] Apr 12 10:38:57.917: INFO: No backend services found
I0412 10:38:59.195] Apr 12 10:38:59.194: INFO: Monitoring glbc's cleanup of gce resources:
I0412 10:38:59.195] k8s-ssl-350024a07559f009-83697e1b6f4cfc12--b4d5aa05f1df8c87 (ssl-certificate)
I0412 10:38:59.195] 
I0412 10:39:07.635] Apr 12 10:39:07.634: INFO: No backend services found
I0412 10:39:08.924] Apr 12 10:39:08.924: INFO: Monitoring glbc's cleanup of gce resources:
I0412 10:39:08.925] k8s-ssl-350024a07559f009-83697e1b6f4cfc12--b4d5aa05f1df8c87 (ssl-certificate)
... for a long time
```

glbc.log
```log
I0412 10:34:39.828468       1 l7.go:611] validating https for e2e-tests-ingress-zqvkk-echomap--b4d5aa05f1df8c87
I0412 10:34:40.122296       1 l7.go:285] Populating ssl cert k8s-ssl-350024a07559f009-83697e1b6f4cfc12--b4d5aa05f1df8c87 for l7 e2e-tests-ingress-zqvkk-echomap--b4d5aa05f1df8c87
I0412 10:34:40.122337       1 l7.go:344] Creating new sslCertificate k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87 for e2e-tests-ingress-zqvkk-echomap--b4d5aa05f1df8c87
I0412 10:34:41.970998       1 l7.go:344] Creating new sslCertificate k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87 for e2e-tests-ingress-zqvkk-echomap--b4d5aa05f1df8c87
E0412 10:34:42.331197       1 l7.go:351] Failed to create new sslCertificate k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87 for e2e-tests-ingress-zqvkk-echomap--b4d5aa05f1df8c87 - googleapi: Error 409: The resource 'projects/k8s-ingress-boskos-17/global/sslCertificates/k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87' already exists, alreadyExists
E0412 10:34:42.331293       1 taskqueue.go:85] Requeuing "e2e-tests-ingress-zqvkk/echomap" due to error: Cert creation failures - k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87 Error:googleapi: Error 409: The resource 'projects/k8s-ingress-boskos-17/global/sslCertificates/k8s-ssl-350024a07559f009-62bfca2bfa77e63f--b4d5aa05f1df8c87' already exists, alreadyExists (ingresses)
```
At which point, the error causes the controller to forget to cleanup the old certificate, thus an orphan.

I don't yet know why the TLS list has the same certificate multiple times. It may be the test jig causing this; however, the controller should handle this situation.

### CODE REVIEW

**Comment by nicksardo** [comment]
I believe the test is flaky because there's a bug in the test jig.
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/ingress_utils.go#L230
The ingress is updated, but doesn't wait for the new certificate to appear. Still not sure why there were identical certs...

File: pkg/loadbalancers/loadbalancers_test.go [hunk 1]
```diff
@@ -160,6 +160,31 @@ func TestCertUpdate(t *testing.T) {
 	verifyCertAndProxyLink(expectCerts, expectCerts, f, t)
 }
 
+// Test that multiple secrets
```
Comment (rramkumar1) [inline pkg/loadbalancers/loadbalancers_test.go]: Finish the comment

File: pkg/loadbalancers/l7.go [hunk 2]
```diff
@@ -317,47 +315,55 @@ func (l *L7) checkSSLCert() error {
...
 	var failedCerts []string
 
 	for _, tlsCert := range l.runtimeInfo.TLS {
 		ingCert := tlsCert.Cert
 		ingKey := tlsCert.Key
-		newCertName := l.namer.SSLCertName(l.Name, tlsCert.CertHash)
+		gcpCertName := l.namer.SSLCertName(l.Name, tlsCert.CertHash)
+
+		if _, exists := newCertMap[gcpCertName]; exists {
+			glog.V(3).Infof("Secret %q had a certificate already created by another secret", tlsCert.Name)
```
Comment (rramkumar1) [inline pkg/loadbalancers/l7.go]: Can we put the name of the other secret in the comment if possible?

**Comment by nicksardo** [comment]
Test fix at https://github.com/kubernetes/kubernetes/pull/62502

**Pull request merged**